yt

```diff
@@ -1,69 +1,191 @@
 #!/bin/bash
-# dependencies: mpv youtube-dl fzf rofi/dmenu
-# search videos and playlists on youtube and play them in mpv, without an API 
-# usage:
-# yt					asks for input in stdin, prompts using fzf
-# yt search query		takes input from the passed arg, prompts using fzf
-# yt -r					takes input and prompts using rofi ($guicmd)
+
+# explain usage
+function usage () {
+    echo "usage: $(basename $0)"
+    echo "    -h help"
+    echo "    -c channels/subscriptions"
+    echo "    -s search"
+    echo "    -g gui option"
+    echo "    nothing use defaults"
+    exit 1
+}
+
+# dont use defaults
+useDefaults="f"
+
+# no args -> use defaults
+if [[ ${#} -eq 0 ]]; then
+    useDefaults="t"
+fi
+
+# available flags
+optstring=":s:c:gh"
 
 defcmd="fzf"
-guicmd="rofi -dmenu -i" #uncomment next line for dmenu
-#guicmd="dmenu -i -l 15"
+defaction="c"
+guicmd="rofi -dmenu -i"
+
+#Defaults
 promptcmd="$defcmd"
-if [ -z "$*" ]; then 
-	echo -n "Search: "
-	read -r query
-else
-	case "$1" in
-		-r) query=$(echo | $guicmd -p "Search: ")
-			promptcmd="$guicmd -p Video:";;
-		*) query="$*";;
-	esac
+action="$defaction"
+isGui="f"
+query=""
+
+# subscription list
+sublist=""
+# example of sublist
+# sublist="williamosman
+# gothamchess
+# thomasrandall"
+# (note that the newline between them is necessary)
+
+# if not using defaults search for flags
+if [[ $useDefaults = "f" ]]; then
+    while getopts ${optstring} arg; do
+        case "${arg}" in
+            s) 
+                # search in youtube for a query
+                action="s"
+                query="${OPTARG}"
+                ;;
+            c) 
+                # search in subscriptions for specific channel
+                action="c"
+                query="${OPTARG}"
+                ;;
+            g) 
+                # set gui mode to true and change the prompt to gui prompt
+                isGui="t"
+                promptcmd="$guicmd"
+                ;;
+            h) 
+                # display help / usage
+                echo "Help incoming!"
+                echo
+                usage
+                ;;
+            \?) 
+                # wrong args -> exit with explanation of usage
+                echo "invalid option: -${OPTARG}"
+                echo
+                usage
+                ;;
+            :) 
+                # missing args -> exit with explanation of usage
+                echo "Option -${OPTARG} needs an argument"
+                echo
+                usage
+                ;;
+
+        esac
+    done
 fi
+
+# if no query is set with flags then ask for one
+if [ -z "$query" ]; then
+    # ask for a channel
+    if [[ $action = "c" ]]; then
+        # if no channels in sub list -> exit
+        if [ -z "$query" ]; then
```

Comment:
I'm just glancing over the code, but should'nt this be $sublist instead?
```suggestion
        if [ -z "$sublist" ]; then
```

Reply:
Yes, hahahaha i copied it from above and forgot to change it.

Reply:
yeah i guessed so 😄
no worries, i'll change that